src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseScoresChartResource.java

```diff
@@ -0,0 +1,89 @@
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.ExerciseScoresChartService;
+import de.tum.in.www1.artemis.web.rest.dto.ExerciseScoresDTO;
+
+/**
+ * Controller to provides endpoints to query the necessary data for the exercise-scores-chart.component.ts in the client
+ */
+@RestController
+public class ExerciseScoresChartResource {
+
+    private final Logger log = LoggerFactory.getLogger(ExerciseScoresChartResource.class);
+
+    private final ExerciseScoresChartService exerciseScoresChartService;
+
+    private final CourseRepository courseRepository;
+
+    private final UserRepository userRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    public ExerciseScoresChartResource(ExerciseScoresChartService exerciseScoresChartService, CourseRepository courseRepository, UserRepository userRepository,
+            AuthorizationCheckService authorizationCheckService) {
+        this.exerciseScoresChartService = exerciseScoresChartService;
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * GET /courses/:courseId/charts/exercise-scores
+     * <p>
+     * This call returns the the information used for the exercise-scores-chart. It will get get the score of
+     * the requesting user,the average score and the best score in course exercises
+     * <p>
+     * Note: Only released course exercises with assessment due date over are considered!
+     * <p>
+     *
+     * @param courseId id of the course for which to get the exercise scores
+     * @return the ResponseEntity with status 200 (OK) and with the exercise scores in the body
+     */
+    @GetMapping(EndpointConstants.COURSE_EXERCISE_SCORES)
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<ExerciseScoresDTO>> getCourseExerciseScores(@PathVariable Long courseId) {
+        log.debug("REST request to get exercise scores for course with id: {}", courseId);
+        Course course = courseRepository.findByIdWithEagerExercisesElseThrow(courseId);
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);
+        // we only consider exercises in which the student had a chance to earn a score (released and due date over)
+        List<ExerciseScoresDTO> exerciseScoresDTOList = exerciseScoresChartService.getExerciseScores(removeNotVisibleAndUnfinishedExercise(course.getExercises()), user);
+        return ResponseEntity.ok(exerciseScoresDTOList);
+    }
+
+    private Set<Exercise> removeNotVisibleAndUnfinishedExercise(Set<Exercise> exercises) {
+        return exercises.parallelStream().filter(Exercise::isVisibleToStudents).filter(Exercise::isAssessmentDueDateOver).collect(Collectors.toSet());
```

Comment:
This looks dangerous, what if an exercise has no assessment due date set (in which case `isAssessmentDueDateOver` returns `true`), but the due date of the exercise has not been reached?

Reply:
I do not understand what you mean? Why would this be "dangerous"? Do you have a suggestion for a fix?

Reply:
2cfc1951315ee9ab7c3d3a4a1402ee3aa3b03e51 Do you think this fixes the problem @FrankeLukas ?

Reply:
That looks better. There would also be the special case of exercises without both due date or assessment due date, but I'm not sure if those will cause problems.

> I do not understand what you mean? Why would this be "dangerous"?

My problem was that unfinished exercises would not get filtered when they have no assessment due date. (I.e. no assessment due date, and a due date in the future => we include an exercise which is still "in progress".)

Reply:
Maybe it would also be good to have a private `isExerciseFinished` method here, so that you can do `return exercises.parallelStream().filter(exercise -> isExerciseFinished(exercise)).collect(Collectors.toSet());` with something like

```
private boolean isExerciseFinished(Exercise exercise) {
    // Exercises not visible to the students are not finished either
    if (!exercise.isVisibleToStudents()) {
        return false;
    }
    
    // If an assessment due date is set, only count the exercise as finished if we passed that date
    if (exercise.getAssessmentDueDate() != null) {
        return ZonedDateTime.now().isAfter(exercise.getAssessmentDueDate());
    }
    
    // If no assessment due date is set, make sure to only count exercises which have a passed due date
    return exercise.getDueDate() != null && ZonedDateTime.now().isAfter(exercise.getDueDate());
}
```

Reply:
Very nice! Looks much cleaner. Did it in e93dbd91abae7cc727edfe2c2d98a4fa5a9234b4